File: Pythonleetcode/Leetcode1588.py
class Solution(object):
    def sumOddLengthSubarrays(self, arr):
        max_len=len(arr)
        step=2
        temp_len=1
        ans=0
        if max_len==0:
            return 0
        else:
            while temp_len<=max_len:
                for i in range(max_len-temp_len+1):
                    ans+=sum(arr[i:i+temp_len])
                temp_len+=step
        return ans

print(Solution().sumOddLengthSubarrays([1,4,2,5,3]))
print(Solution().sumOddLengthSubarrays([1,2]))

File: CppPractice/Project1/Project1/Leetcode1588.cpp
#include <string>

#include <vector>

#include <iostream>

#include <stdlib.h>

#include <time.h>

#include <sstream>

#include <algorithm>

#include <map>

using namespace std;



class Solution {
public:
    int sumOddLengthSubarrays(vector<int>& arr) {
        int temp_len = 1;
        int ans = 0;
        while (temp_len <= arr.size())
        {
            for (int i = 0; i < arr.size() - temp_len + 1; i++)
            {
                for (int j = 0; j < temp_len; j++)
                {
                    ans += arr[i + j];
                }

            }
            temp_len += 2;

        }
        return ans;



    }
};